test/OpenTelemetry.Extensions.Hosting.Tests/HostingExtensionsTests.cs

```diff
@@ -76,44 +81,78 @@ public void AddOpenTelemetryTracerProvider_HostBuilt_OpenTelemetrySdk_Registered
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.Configure<TracerProviderBuilder>((builder) =>
+                {
+                    builder.AddInstrumentation(() =>
+                    {
+                        instrumentationCallBackCount++;
+                        return testInstrumentation;
+                    });
+                });
             });
 
-            var serviceProvider = services.BuildServiceProvider();
-
-            var tracerFactory = serviceProvider.GetRequiredService<TracerProvider>();
-            Assert.NotNull(tracerFactory);
-
-            Assert.False(testInstrumentation.Disposed);
-
-            serviceProvider.Dispose();
-
-            Assert.True(testInstrumentation.Disposed);
+            var host = builder.Build();
+            await host.StartAsync();
+            Assert.Equal(1, instrumentationCallBackCount);
+            await host.StopAsync();
+            host.Dispose();
         }
 
         [Fact]
-        public void AddOpenTelemetryTracerProvider_BadArgs_NullServiceCollection()
+        public async Task AddOpenTelemetryTracingE2EDemo()
         {
-            ServiceCollection services = null;
-            Assert.Throws<ArgumentNullException>(() => services.AddOpenTelemetryTracing());
-            Assert.Throws<ArgumentNullException>(() =>
-            services.AddOpenTelemetryTracing((provider, builder) =>
+            var testProcessor = new TestActivityProcessor();
+            var resource = Resources.Resources.CreateServiceResource("serviceName");
+            testProcessor.StartAction = (act) => Assert.Equal("MyActivityName", act.DisplayName);
+            testProcessor.EndAction = (act) => Assert.Equal(resource, act.GetResource());
+            var builder = new HostBuilder().ConfigureServices(services =>
             {
-                builder.AddInstrumentation(() => provider.GetRequiredService<TestInstrumentation>());
-            }));
+                // Add OpentelemetryTracing.
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+
+                // Configure TracerProviderBuilder
+                services.Configure<TracerProviderBuilder>((builder) => builder.SetSampler(new AlwaysOnSampler()));
+
+                // Keep configuring TracerProviderBuilder
+                services.Configure<TracerProviderBuilder>((builder) => builder.AddProcessor(testProcessor));
+
+                // Keep configuring TracerProviderBuilder
+                services.Configure<TracerProviderBuilder>((builder) => builder.SetResource(resource));
+
+                // Keep configuring TracerProviderBuilder
+                services.Configure<TracerProviderBuilder>((builder) => builder.AddSource(TestActivitySourceName));
```

Comment:
to demo - how to build the TracerProviderBuilder from any place. The order in which Configure is called is the order in which they are called.